examples/mongocxx/exception.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+#include <iostream>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/exception/bulk_write_exception.hpp>
+#include <mongocxx/exception/error_code.hpp>
+#include <mongocxx/exception/server_error_code.hpp>
+#include <mongocxx/exception/logic_error.hpp>
+#include <mongocxx/exception/operation_exception.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/uri.hpp>
+
+using bsoncxx::builder::stream::close_document;
+using bsoncxx::builder::stream::document;
+using bsoncxx::builder::stream::finalize;
+using bsoncxx::builder::stream::open_document;
+
+int main(int, char**) {
+    mongocxx::instance inst{};
+    mongocxx::client conn{mongocxx::uri{}};
+
+    // @begin: cpp-logic-error
+    // Using an uninitialized collection throws a mongocxx::logic_error.
+    // A mongocxx::logic_error is-a mongocxx::exception is-a std::system_error.
+    mongocxx::collection coll;
+    try {
+        coll.name();
+    } catch (const mongocxx::logic_error& e) {
+        std::cout << "Using an uninitialized collection throws:" << std::endl;
+
+        if (e.code().category() != mongocxx::error_category()) {
+            return EXIT_FAILURE;
+        }
+
+        // We can compare the error_code to a known mongocxx::error_code.
+        if (e.code() != mongocxx::error_code::k_invalid_collection_object) {
+            return EXIT_FAILURE;
+        }
+
+        std::cout << e.what() << std::endl << std::endl;
+    }
+    // @end: cpp-logic-error
+
+    // @begin: cpp-operation-exception
+    // Renaming a collection that does not exist throws a mongocxx::operation_exception.
+    // A mongocxx::operation_exception is-a mongocxx::exception is-a std::system_error.
+    coll = conn["test"]["coll"];
+    coll.drop();
+    try {
+        coll.rename("coll2");
+    } catch (const mongocxx::operation_exception& e) {
+        std::cout << "Renaming a collection that does not exist throws:" << std::endl;
+
+        // We expect this error to arise from the server
+        if (e.code().category() != mongocxx::server_error_category()) {
+            return EXIT_FAILURE;
+        }
+
+        // We can compare the error_code to a known server side error code number
+        if (e.code().value() != 26) {
```

Comment:
Perhaps include a link to [error_codes.err](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.err) to explain where 26 came from, and to show users the other codes they might use